libraries/AC_Sprayer/AC_Sprayer.cpp

```diff
@@ -107,8 +108,53 @@ void AC_Sprayer::stop_spraying()
 {
     SRV_Channels::set_output_limit(SRV_Channel::k_sprayer_pump, SRV_Channel::Limit::MIN);
     SRV_Channels::set_output_limit(SRV_Channel::k_sprayer_spinner, SRV_Channel::Limit::MIN);
+}
+
+// update sprayer pwm
+void AC_Sprayer::update_servos(const float ground_speed)
+{
+    if (!_flags.spraying) {
+        stop_spraying();
+        return;
+    }
+
+    // set pwm according to ground speed
+    float pos = ground_speed * _pump_pct_1ms;
+    pos = MAX(pos, 100 *_pump_min_pct); // ensure min pump speed
+    pos = MIN(pos,10000); // clamp to range
```

Comment:
Shouldn't this clamp to the servo range?

Reply:
This clamps the pos to a maximum value and then the `pos` value along with it's minimum and maximum limit is passed to `SRV_Channels::move_servo` method. This method scales the value of `pos` as per the real minimum and maximum value of servo range and send the pwm output to the servo channel. That is, if the value of pos is 50% on the scale of 0 to 10000, then the pwm output is also 50% on the scale of servo's minimum and maximum pwm.

Also, this was pre existing. Please let me know if I am getting something wrong. I also did test it and saw the plot of pwm values of servo channel and the values looked good.